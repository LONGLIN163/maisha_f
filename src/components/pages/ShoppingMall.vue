<template>
    <div>
        <div class="search-bar">
            <van-row>
                <van-col span="3">
                    <img :src="locationIcon" alt="" width="70%" class="location-icon">
                </van-col>
                <van-col span="16">
                    <input type="text" class="search-input"/>
                </van-col>
                <van-col span="5">
                    <van-button size="mini">search</van-button>
                </van-col>
            </van-row>
        </div>

        <!-- slider -->
        <div class="swiper-area">
            <van-swipe :autoplay="1000">
                <van-swipe-item v-for="(banner,index) in bannerPicArray" :key="index">
                    <img v-lazy="banner.image" width="100%"/>
                </van-swipe-item>
            </van-swipe>
        </div>

        <!-- types -->
        <div class="type-bar">
            <div  v-for="(item,index) in category" :key="index" >
                    <img v-lazy="item.image" width="90%" />
                    <span>{{item.mallCategoryName}}</span> 
            </div>
        </div>

        <!--AD banner area-->
        <div class="ad-banner">
            <img v-lazy="adBanner.PICTURE_ADDRESS" width="100%">
        </div>
    </div>
</template>

<script>
    import axios from "axios"
    export default {
        data() {
            return {
                msg: 'haha',
                locationIcon:require("../../assets/images/location.png"),
                bannerPicArray:[],
                category:[],
                adBanner:{}

            }
        },
        created(){
            axios({
                url:"https://mocki.io/v1/56645569-6ae1-4faa-a9cc-25a88f18dd82",
                method:"GET",

            }).then(res=>{
                console.log(res)
                if(res.status==200){
                    this.category = res.data.data.category;
                    this.adBanner = res.data.data.advertesPicture
                    this.bannerPicArray = res.data.data.slides
                }
            }).catch(err=>{
                console.log(err)
            })
        }

    }
</script>



<style scoped>
  .search-bar{
      height: 2.2rem;
      background-color: #e5017d;
      line-height:2.2rem;
      overflow: hidden;

  }
  .search-input{
      width:100%;
      height: 1.3rem;
      border-top:0px;
      border-left:0px;
      border-right:0px;
      border-bottom: 1px solid rgb(255, 255, 255) !important ; 
      background-color: #e5017d;
      color:#fff;
  }
  .location-icon{
      padding-top: .2rem;
      padding-left: .3rem;
  }
  .swiper-area{
      width:20rem;
      max-height: 15rem;
      overflow: hidden;
  }
   .type-bar{
      background-color: #fff;
      margin:0 .3rem .3rem .3rem;
      border-radius: .3rem;
      font-size:14px;
      display: flex;
      flex-direction:row;
      flex-wrap:nowrap;
  }
  .type-bar div{
      padding: .3rem;
      font-size: 12px;
      text-align: center;
  }
</style>