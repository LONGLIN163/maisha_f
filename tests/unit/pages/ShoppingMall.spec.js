import { shallowMount,mount, createLocalVue} from '@vue/test-utils'
import ShoppingMall from '@/components/pages/ShoppingMall.vue'

import Vue from 'vue'

const Vant=require("vant")

const {Lazyload}=Vant

Vue.use(Lazyload)

const localVue = createLocalVue();
localVue.use(Vant); 

describe('ShoppingMall', () => {
  const wrapper = mount(ShoppingMall,{
    localVue,
    Lazyload
  })

  it('renders ShoppingMall', () => {
    expect(wrapper).toMatchSnapshot()
  })
})