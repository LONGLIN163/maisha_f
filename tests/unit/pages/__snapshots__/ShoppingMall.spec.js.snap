// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingMall should renders ShoppingMall 1`] = `
<div>
  <div class="search-bar">
    <div class="van-row">
      <div class="van-col van-col--3"><img src="[object Object]" alt="" width="70%" class="location-icon"></div>
      <div class="van-col van-col--16"><input type="text" data-test="search-input" class="search-input"></div>
      <div class="van-col van-col--5"><button data-search="Search" class="van-button van-button--default van-button--mini">
          <div class="van-button__content"><span class="van-button__text">Search</span></div>
        </button></div>
    </div>
  </div>
  <div class="swiper-area">
    <div class="van-swipe">
      <div class="van-swipe__track" style="transition-duration: 500ms; transform: translateX(0px);"></div>
    </div>
  </div>
  <div class="type-bar"></div>
  <div class="ad-banner"><img width="100%" data-src="undefined" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
  <div class="recommend-area">
    <div class="recommend-title">
      Recommend
    </div>
    <div class="recommend-body">
      <div class="swiper-container swiper-container-initialized swiper-container-horizontal">
        <div class="swiper-wrapper" style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"></div> <span class="swiper-notification" aria-live="assertive" aria-atomic="true"></span>
      </div>
    </div>
  </div>
  <div>
    <div class="floor">
      <div class="floor-title"></div>
      <div class="floor-anomaly">
        <div class="van-row">
          <div class="floor-one van-col van-col--12"><img alt=""></div>
          <div class="floor-two van-col van-col--12">
            <div>
              <div><img></div>
              <div><img></div>
            </div>
          </div>
        </div>
      </div>
      <div class="floor-rule"></div>
    </div>
  </div>
  <div>
    <div class="floor">
      <div class="floor-title"></div>
      <div class="floor-anomaly">
        <div class="van-row">
          <div class="floor-one van-col van-col--12"><img alt=""></div>
          <div class="floor-two van-col van-col--12">
            <div>
              <div><img></div>
              <div><img></div>
            </div>
          </div>
        </div>
      </div>
      <div class="floor-rule"></div>
    </div>
  </div>
  <div>
    <div class="floor">
      <div class="floor-title"></div>
      <div class="floor-anomaly">
        <div class="van-row">
          <div class="floor-one van-col van-col--12"><img alt=""></div>
          <div class="floor-two van-col van-col--12">
            <div>
              <div><img></div>
              <div><img></div>
            </div>
          </div>
        </div>
      </div>
      <div class="floor-rule"></div>
    </div>
  </div>
  <div class="hot-area">
    <div class="hot-title">HOT</div>
    <div class="hot-goods">
      <div role="feed" class="van-list">
        <div class="van-row">
          <div class="hotgoodsitem van-col van-col--12" style="padding-right: 10px;">
            <div class="goods-info">
              <div class="goods-image"><img width="90%" data-src="undefined" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
              <div class="goods-name"></div>
              <div class="goods-price">0.00 €</div>
            </div>
          </div>
          <div class="hotgoodsitem van-col van-col--12" style="padding-left: 10px;">
            <div class="goods-info">
              <div class="goods-image"><img width="90%" data-src="undefined" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
              <div class="goods-name"></div>
              <div class="goods-price">0.00 €</div>
            </div>
          </div>
        </div>
        <div class="van-list__placeholder"></div>
      </div>
    </div>
  </div>
</div>
`;
